lib/events.js

```diff
@@ -497,6 +497,17 @@ function unwrapListeners(arr) {
 
 function once(emitter, name) {
   return new Promise((resolve, reject) => {
+    if (typeof emitter.addEventListener === 'function') {
+      // EventTarget does not have `error` event semantics like Node
+      // EventEmitters, we do not listen to `error` events here.
+      emitter.addEventListener(
```

Comment:
@mcollina ... what do you think on this? The lack of listening to error here may cause some memory leaks in some situations. Given the exploration you've done on the async iterators and handling of async functions as event handlers, I'd love your opinion on the approach here.

Reply:
The original iteration of this had some discussion on error handling. This semantic was actually discussed with Matteo live during the Berlin summit and this comment was typed with him sitting nearby :] 

I am honestly not sure this is the right approach but the web does not have error semantics like Node so this is the best we came up with then.

Reply:
Nice, ok. Carry on then!